lib/knapsack_pro/formatters/time_tracker.rb

```diff
@@ -0,0 +1,162 @@
+
+      def example_started(_notification)
+        @before_all = now - @time_all if @before_all == 0.0
+        @time_each = now
+      end
+
+      def example_finished(notification)
+        record_example(@group, notification.example, @time_each)
+        @time_all = now
+      end
+
+      def example_group_finished(notification)
+        return unless top_level_group?(notification.group)
+
+        add_hooks_time(@group, @before_all, now - @time_all)
+        @subset = merge(@subset, @group)
+        @before_all = 0.0
+        @group = {}
+      end
+
+      # Called at the end of each subset
+      def stop(_notification)
+        @queue = merge(@queue, @subset)
+        @subset = {}
+        @subset_started = now
+      end
+
+      def queue(scheduled_paths)
+        recorded_paths = @queue.values.map do |example|
+          example[:path].match(/\A(.*?)(?:\[([\d\s:,]+)\])?\z/).captures.first
+        end
+
+        missing = (scheduled_paths - recorded_paths).each_with_object({}) do |path, object|
+          object[path] = { path: path, time_execution: 0.0 }
+        end
+
+        merge(@queue, missing).values.map do |example|
+          example.transform_keys(&:to_s)
+        end
+      end
+
+      def subset
+        @subset.values.map do |example|
+          example.transform_keys(&:to_s)
+        end
+      end
+
+      def duration
+        now - @suite_started
+      end
+
+      def subset_duration
+        now - @subset_started
+      end
+
+      def unexecuted_test_files(scheduled_paths)
+        pending_paths = (@queue.values + @subset.values)
+          .filter { |example| example[:time_execution] == 0.0 }
+          .map { |example| example[:path] }
+
+        not_run_paths = scheduled_paths -
+          (@queue.values + @subset.values)
+          .map { |example| example[:path] }
+
+        pending_paths + not_run_paths
+      end
+
+      private
+
+      def top_level_group?(group)
+        group.metadata[:parent_example_group].nil?
+      end
+
+      def add_hooks_time(group, before_all, after_all)
+        group.each do |_, example|
+          next if example[:time_execution] == 0.0
+          example[:time_execution] += before_all + after_all
+        end
+      end
+
+      def record_example(accumulator, example, started_at)
+        path = path_for(example)
+        time_execution = time_execution_for(example, started_at)
+
+        if accumulator.key?(path)
+          accumulator[path][:time_execution] += time_execution
+        else
+          accumulator[path] = { path: path, time_execution: time_execution }
+        end
+      end
+
+      def path_for(example)
+        file = file_path_for(example)
+        return "UNKNOWN_PATH" if file == ""
+        path = rspec_split_by_test_example?(file) ? example.id : file
+        KnapsackPro::TestFileCleaner.clean(path)
+      end
+
+      def rspec_split_by_test_example?(file)
+        return false unless KnapsackPro::Config::Env.queue_recording_enabled?
```

Comment:
```suggestion
```

I think this line is not needed and it would solve the issue https://github.com/KnapsackPro/knapsack_pro-ruby/pull/229#issuecomment-1830474240